image-types/README.md

```diff
@@ -0,0 +1,8 @@
+# OSBuild Composer Image Types
+
+This directory contains high-level descriptions of all image types that
+**osbuild-composer** generates, grouped by operating system.
```

Comment:
Why would we group them by OS? And are the definitions different across the different operating systems? If the definition is mostly the same, then having one place where the definition is defined would help avoid duplication. And the definition can include what's unique per OS, which might make it easier to review and maintain the definitions.

Reply:
Yes, different operating systems can have different packages, kernel parameters, initramdisk modules etc. If I am not mistaken, that is the case in Azure. But otherwise it should be mostly the same.

Reply:
There might be some overlap indeed, but in general they're different for each operating system. Most importantly, the reasoning might be. Also, different operating systems call their images different things (Fedora Cloud Base vs RHEL KVM Guest Image, for example).

These documents will also grow as new releases for these come out. I didn't want to make them too complex by conflating different releases and different operating systems in the same document.